Makefile

```diff
@@ -0,0 +1,45 @@
+LLVM_PROJECT_DIR := llvm-project
+BUILD_DIR := $(LLVM_PROJECT_DIR)/build
+TARGET_OS := $(shell uname | tr '[:upper:]' '[:lower:]')
+TARGET_ARCH := $(shell uname -m)
+NUM_CORES := $(shell nproc 2>/dev/null || sysctl -n hw.ncpu)
+
+.PHONY: all clone init checkout configure build show-lib clean
+
+all: init checkout configure build show-lib
+
+clone:
```

Comment:
Instead of cloning the repo, we can download tagged sources. This is much faster (but also less flexible)

Reply:
I gave this a try, and it seems like it is a bit slower on my machine at least.

The main issue is that when you download the sources from the release, you have to download the entire repo. This is because the CMake expects there at least to be the top level `cmake` directories available from the stuff we don't use (`clang`, `llvm` etc).

The git method selectively downloads just the `cmake` directories, and `compiler-rt`, instead of everything, which makes the downloading and extracting faster.

Do you have an idea of how we could do this while maintaining the speed?

To check out what I did, see this branch:

https://github.com/cjappl/rtsan-libs/tree/try_download

Reply:
Interesting. Maybe there is something else at play here, but my action using download takes a minute:

https://github.com/realtime-sanitizer/RTSanStandaloneSwift/actions/runs/14124620231/job/39570957031

while clone takes 3 minutes.

Reply:
It is entirely possible this is my crappy internet. I will run it on the build bot to compare apples to apples before dismissing it entirely.

I do agree that using the official released sources is nice if the time is in the ballpark, the `git` part of this makefile is definitely more complex than I'd like.

Will report back

Reply:
@jcavar - oh yeah, downloading the sources is WAY faster.

See for yourself here: 

Git clone method - https://github.com/cjappl/rtsan-libs/actions/runs/14206157744
ubuntu - 5 min
macos - 5.5 min

Download method - https://github.com/cjappl/rtsan-libs/actions/runs/14206157744
ubuntu - 0.5 min
macos - 2.5 min

Switched this review to downloading and extracting the sources. Would you please take another look to make sure I didn't mess anything up? Going to re-request a review from you